File: beginner/C_Kochan/ch19/p1.m
// Example 19.2: Program to work with fractions - Objective C version
// However this program is extremely outdated since the objc/Object.h
//has been deprecated since October, 2007.
// Hence, the program does not work!
#import <stdio.h>
#import <objc/Object.h>

// ------- @interface section -------
@interface Fraction: Object
{
  int numerator;
  int denominator;
}
-(int) numerator;
-(int) denominator;
-(void)  setNumerator: (int) n;
-(void)  setDenominator: (int) d;
-(void)  print;

@end

// ------- @implementation section ------
@implementation Fraction;
// getters
-(int) numerator
{
  return numerator;
}

-(int) denominator
{
  return denominator;
}

// setters
-(void) setNumerator: (int) num
{
  numerator = num;
}

-(void) setDenominator: (int) denom
{
  denominator = denom;
}

// other
-(void) print
{
  printf("The value of the fraction is %i/%i\n", numerator,
  denominator);
}

@end

//--------- program section ------------

int main(void) {
  Fraction *myFrac;

  myFrac = [Fraction new];

  [myFrac setNumerator: 1];
  [myFrac setDenominator: 3];

  printf("The numerator is %i, and the denominator is %i\n",
[myFrac numerator], [myFrac denominator]);
  [myFrac print];

  [myFrac free];

  return 0;
}

File: beginner/C_Kochan/ch19/p1.cs
// Program 19.4: Simple program with fractions -- C# version
using System;

class Fraction
{
  private int numerator;
  private int denominator;

  public int Numerator
  {
    set
    {
      numerator = value;
    }
    get
    {
      return numerator;
    }
  }

  public int Denominator
  {
    set
    {
      denominator = value;
    }
    get
    {
      return denominator;
    }
  }

  public void print()
  {
    Console.WriteLine("The value of the fraction is {0}/{1}.\n",
    numerator, denominator);
  }
}

class example
{
  public static void Main()
  {
    Fraction myFrac = new Fraction();

    myFrac.Numerator = 1;
    myFrac.Denominator = 3;

    myFrac.print();
  }
}
